### PR TITLE
InlineHelp: Make focus on the search input element once open

### PR DESCRIPTION
**Pull request by retrofox**
#### Changes proposed in this Pull Request

This PR focuses on the input element of the search card. I've had to add a timeout since there is a race condition that makes the flow lose the focus.
Also, it rewrites the component using a function instead of a class inherited from the react Component one.

#### Testing instructions

* Apply these changes
* Click on the FAB icon on order to open the popover
* Confirm that it focuses on the input search element.

inline-help-focuses

Fixes https://github.com/Automattic/wp-calypso/issues/43958

### CODE REVIEW

File: client/blocks/inline-help/inline-help-search-card.jsx [hunk 1]
```diff
@@ -139,9 +158,9 @@ const mapStateToProps = ( state, ownProps ) => ( {
 } );
 const mapDispatchToProps = {
 	track: recordTracksEvent,
-	setInlineHelpSearchQuery,
-	selectNextResult,
-	selectPreviousResult,
+	seSearchQuery: setInlineHelpSearchQuery,
```
Comment (frontdevde) [inline client/blocks/inline-help/inline-help-search-card.jsx:161]: Should this be `setSearchQuery`? Probably missing a `t`

Comment (frontdevde) [reply client/blocks/inline-help/inline-help-search-card.jsx]: Looks like, @cpapazoglou spotted it as well and just pushed a change 👌

Comment (retrofox) [reply client/blocks/inline-help/inline-help-search-card.jsx]: outch

File: client/blocks/inline-help/inline-help-search-card.jsx [hunk 2]
```diff
@@ -30,31 +30,43 @@ import { SUPPORT_TYPE_ADMIN_SECTION } from './constants';
...
+		const inputElement = cardRef.current.searchInput;
+		if ( ! inputElement ) {
+			return;
+		}
+		const timerId = setTimeout( () => inputElement.focus(), 0 );
 
-		this.searchHelperHandler = this.searchHelperHandler.bind( this );
-	}
+		return () => window.clearTimeout( timerId );
+	}, [ cardRef, location ] );
```
Comment (cpapazoglou) [inline client/blocks/inline-help/inline-help-search-card.jsx:67]: Would you consider rewriting as follows?
```
useEffect( () => {
    const inputElement = cardRef.current?.searchInput;
    // Focuses only in the popover.
    if ( location !== 'inline-help-popover' || ! inputElement ) {
        return;
    }

    const timerId = setTimeout( () => inputElement.focus(), 0 );

    return () => window.clearTimeout( timerId );
}, [ cardRef, location ] );
```
`cardRef.current` always exists as per react api

**Comment by retrofox** [comment]
```
 ! [rejected]        update/inline-help-keep-open -> update/inline-help-keep-open (fetch first)
error: failed to push some refs to 'git@github.com:Automattic/wp-calypso.git'
hint: Updates were rejected because the remote contains work that you do
hint: not have locally. This is usually caused by another repository pushing
```

It explains everything :-D. Thanks, @cpapazoglou ✋

File: client/blocks/inline-help/inline-help-search-card.jsx (hunk 2 above)
Comment (retrofox) [reply client/blocks/inline-help/inline-help-search-card.jsx:60]: 5c2d8fbf9b080563f900d69f4ebfa04f94068e37

**Pull request merged**